pkg/document/crdt/rht.go

```diff
@@ -48,7 +48,7 @@ func (n *RHTNode) Remove(removedAt *time.Ticket) {
 }
 
 func (n *RHTNode) isRemoved() bool {
-	return n.removedAt != nil
+	return n.removedAt != nil && n.updatedAt != nil && n.removedAt.Compare(n.updatedAt) == 0
```

Comment:
Why is this comparison(`&& n.updatedAt != nil && n.removedAt.Compare(n.updatedAt) == 0`) necessary?

Reply:
Since the removed elements always guarantee the invariant `updatedAt = removedAt`, it is essential to check this condition in the `isRemoved`. Although I considered modifying only `removedAt` in the `Remove`, but this approach would be inconvenient, because we have to compare both `updatedAt` and `removedAt` with `executedAt` in the `RHT.Set`.

Reply:
Then, is it possible to express whether to remove a node with just a flag(`isRemoved`)?

Reply:
I think it's a good idea. I'll add the `isRemoved` flag and reimplement it.